File: app/src/main/java/com/example/pocedex/data/CommentAndFavorite.java
package com.example.pocedex.data;


import android.os.Parcel;
import android.os.Parcelable;

public class CommentAndFavorite implements Parcelable {

    private String name;
    private int id;
    private String url;
    private boolean is_favorite = false;
    private String comment = null;

    public CommentAndFavorite() {
    }

    public CommentAndFavorite(String name, int id, String url, boolean is_favorite, String comment) {
        this.name = name;
        this.id = id;
        this.url = url;
        this.is_favorite = is_favorite;
        this.comment = comment;
    }


    public boolean equals(int id) {
        if (this.id == id) {
            return true;
        }
        return false;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setUrl(String link) {
        this.url = link;
    }

    public String getUrl() {
        return url;
    }

    public void setIsFav(boolean isfavorite) {
        is_favorite = isfavorite;
    }

    public boolean getIsFav() {
        return is_favorite;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public String getComment() {
        return comment;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Parcelable.Creator<CommentAndFavorite> CREATOR = new Parcelable.Creator<CommentAndFavorite>() {

        @Override
        public CommentAndFavorite createFromParcel(Parcel source) {
            String name = source.readString();
            int id = source.readInt();
            String url = source.readString();
            boolean is_fav;
            if (source.readInt() == 1)
                is_fav = true;
            else is_fav = false;
            String comment = source.readString();
            return new CommentAndFavorite(name, id, url, is_fav, comment);
        }

        @Override
        public CommentAndFavorite[] newArray(int size) {
            return new CommentAndFavorite[size];
        }
    };

    @Override
    public void writeToParcel(Parcel parcel, int flags) {
        parcel.writeString(name);
        parcel.writeInt(id);
        parcel.writeString(url);
        if (is_favorite) {
            parcel.writeInt(1);
        } else {
            parcel.writeInt(0);
        }
        parcel.writeString(comment);
    }

}

File: app/src/main/java/com/example/pocedex/data/CommentAndFavorite.kt
package com.example.pocedex.data

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.Gson
import kotlinx.android.parcel.TypeParceler

data class CommentAndFavorite(var pokemon: Pokemon=Pokemon(), var is_favorite: Boolean = false, var comment: String? = "") : Parcelable {


    fun alreadyExists(id: Int): Boolean {
        return this.pokemon.id == id
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<CommentAndFavorite> {
        override fun createFromParcel(source: Parcel): CommentAndFavorite {
            val gson = Gson()
            val pokemon: Pokemon = gson.fromJson(source.readString(), Pokemon::class.java)
            val isFav: Boolean=(source.readInt() == 1)
            val comment = source.readString()
            return CommentAndFavorite(pokemon, isFav, comment)
        }

        override fun newArray(size: Int): Array<CommentAndFavorite?> {
            return arrayOfNulls(size)
        }
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        val gson = Gson()
        val jsonString = gson.toJson(pokemon)
        parcel.writeString(jsonString)
        if (is_favorite) {
            parcel.writeInt(1)
        } else {
            parcel.writeInt(0)
        }
        parcel.writeString(comment)
    }

}
